doct.el

```diff
@@ -1282,6 +1283,87 @@ Otherwise, add to front."
   (let ((args (list (doct declarations) (copy-tree list))))
     (apply #'append (if append (nreverse args) args))))
 
+
+;; A structure for captures
+;; PARENTS is a list of strings which are the names of the capture's parents
+;; CHILD is t if the capture is a child of another and nil if it isn't
+;; DECLARATION is the declaration of the capture in doct
+(cl-defstruct doct--capture
+  parents
+  child
+  declaration)
+
+;; The capture namespace. Hash-keys are strings.
+(defvar doct--capture-namespace (make-hash-table :test 'equal))
+
+(defun doct--capture-boundp (name)
+  "Return t if NAME is bound to a capture and nil if it isn't."
+  (and (gethash name doct--capture-namespace) t))
+
+
+(defun doct--compute-children (name)
+  "Search the doct--capture-namespace for the children of the capture NAME.
+Return them as a list."
+  (cl-loop for k being each hash-key of doct--capture-namespace
+           using (hash-value c)
+           appending
+           (when (cl-member name (doct--capture-parents c) :test #'equal)
+             (list k))))
+
+(defun doct--make-declaration (name)
```

Comment:
We can drop the words `make` and `compute` in favor of what the functions return. e.g.

`doct--declaration-children` and `doct--declaration`